extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/AddContainerPortDecorator.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.kubernetes.deployment;
+
+import io.dekorate.deps.kubernetes.api.model.ContainerBuilder;
+import io.dekorate.kubernetes.annotation.Protocol;
+import io.dekorate.kubernetes.config.Port;
+import io.dekorate.kubernetes.decorator.AddSidecarDecorator;
+import io.dekorate.kubernetes.decorator.ContainerDecorator;
+import io.dekorate.kubernetes.decorator.Decorator;
+import io.dekorate.kubernetes.decorator.ResourceProvidingDecorator;
+
+/**
+ * Adapted from Dekorate's {@link io.dekorate.kubernetes.decorator.AddPortDecorator} in order to
+ * address the limitation that the port was not being added (due to missing metadata)
+ */
+class AddContainerPortDecorator extends Decorator<ContainerBuilder> {
+
+    private final Port port;
+
+    AddContainerPortDecorator(Port port) {
+        this.port = port;
+    }
+
+    @Override
+    public void visit(ContainerBuilder containerBuilder) {
+        containerBuilder.addNewPort()
```

Comment:
What happens if the container you add has a port named after the port you are trying to add?

Reply:
That probably won't go down well, but I think it's rather unlikely to happen, no?